includes/classes/class-fixinsecurecontent.php

```diff
@@ -0,0 +1,317 @@
+		$post_content = $current_post->post_content;
+		$urls         = $this->parse_content_for_insecure_urls( $post_content );
+		$count        = 0;
+
+		// Check if post content has insecure URLs.
+		if ( empty( $urls ) ) {
+			// translators: %d The Post ID being scanned for insecure content.
+			$message = sprintf( __( 'No insecure content URL found in post %d', 'insecure-content-warning' ), $post_id );
+			if ( defined( 'WP_CLI' ) && WP_CLI ) {
+				WP_CLI::warning( $message );
+			} else {
+				$this->warning_count[] = array( __( 'Warning', 'insecure-content-warning' ) => $message );
+			}
+
+			if ( $this->dry_run ) {
+				// translators: %d The Post ID being scanned for insecure content.
+				$this->fixed_post_count[] = array( __( 'URL(s) fixed summary', 'insecure-content-warning' ) => sprintf( __( '0/0 URL(s) will be fixed in post %d', 'insecure-content-warning' ), $post_id ) );
+			} else {
+				// translators: %d The Post ID being scanned for insecure content.
+				$this->fixed_post_count[] = array( __( 'URL(s) fixed summary', 'insecure-content-warning' ) => sprintf( __( '0/0 URL(s) fixed in post %d', 'insecure-content-warning' ), $post_id ) );
+			}
+
+			return '';
+		}
+
+		foreach ( $urls as $url ) {
+			if ( $this->does_secure_content_exist( $url ) ) {
+				$ssl_url      = preg_replace( '/^http:/i', 'https:', $url );
+				$post_content = str_replace( $url, $ssl_url, $post_content );
+				$count ++;
+			}
+		}
+
+		if ( false === $this->dry_run ) {
+			// Update post content with HTTPS URLs.
+			wp_update_post(
+				array(
+					'ID'           => $post_id,
+					'post_content' => $post_content,
+				)
+			);
+		}
+
+		if ( $this->dry_run ) {
+			// translators: 1: Number of insecure URLs updated, 2: Number of insecure URLs in post, 3: The post ID.
+			$dry_run_message = sprintf( __( '%1$d/%2$d insecure URLs will be fixed in post %3$d.', 'insecure-content-warning' ), $count, count( $urls ), $post_id );
+			if ( defined( 'WP_CLI' ) && WP_CLI ) {
+				WP_CLI::debug( $dry_run_message );
+			}
+			$this->fixed_post_count[] = array( 'URL(s) fixed summary' => $dry_run_message );
+		} else {
+			// translators: 1: Number of insecure URLs updated, 2: Number of insecure URLs in post, 3: The post ID.
+			$success_message = sprintf( __( '%1$d/%2$d insecure URLs fixed in post %3$d.', 'insecure-content-warning' ), $count, count( $urls ), $post_id );
+			if ( defined( 'WP_CLI' ) && WP_CLI ) {
+				WP_CLI::debug( $success_message );
+			}
+			$this->fixed_post_count[] = array( 'URL(s) fixed summary' => $success_message );
+		}
+	}
+
+	/**
+	 * Check if a SSL version of the URL exists.
+	 *
+	 * @param string $url URL to check for SSL version.
+	 *
+	 * @return bool
+	 */
+	protected function does_secure_content_exist( $url ) {
+		// Check if a https version of the URL exists.
+		$secure_version_exists = false;
+		$ssl                   = preg_replace( '/^http:/i', 'https:', $url );
+		$response              = wp_remote_get( $ssl );
```

Comment:
**maybe**, _maybe_, maybe???

[Per MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/HEAD), this should tell you whether the file exists without having to download the full content. It should return the HTTP headers as-though you were making a GET request.

```suggestion
		$response              = wp_remote_head( $ssl );
```